src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,287 @@
+on the dst switch"() {
+        given: "A round trip latency ISL"
+        Isl roundTripIsl
+        def srcSwToDeactivate = topology.activeSwitches.find { sw ->
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
+                roundTripIsl = topology.getRelatedIsls(sw).find {
+                    it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                roundTripIsl
+            }
+        } ?: assumeTrue("Wasn't able to find a switch with suitable link", false)
+        def dstSw = roundTripIsl.dstSwitch
+
+        and: "Round trip status is ACTIVE for the given ISL in both directions"
+        [roundTripIsl, roundTripIsl.reversed].each { isl ->
+            assert database.getIslRoundTripStatus(isl) == IslStatus.ACTIVE
+        }
+
+        when: "Deactivate the src switch"
+        lockKeeper.knockoutSwitch(srcSwToDeactivate)
+        def isSrcSwDeactivated = true
+        // it takes more time to DEACTIVATE a switch via the 'knockoutSwitch' method on the stage env
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip ISL is still DISCOVERED (the system uses round trip latency for ISL alive confirmation)"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.DISCOVERED
+
+        and: "Round trip status for forward direction is not available and ACTIVE in reverse direction"
+        !database.getIslRoundTripStatus(roundTripIsl)
+        database.getIslRoundTripStatus(roundTripIsl.reversed) == IslStatus.ACTIVE
+
+        when: "Delete ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.deleteSwitchRules(dstSw.dpId, DeleteRulesAction.REMOVE_ROUND_TRIP_LATENCY)
+        def isRoundTripRuleDeleted = true
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.size() == 1
+        }
+
+        then: "The round trip latecny ISL is FAILED"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        and: "Round trip status is not available for the given ISL in both directions"
+        [roundTripIsl, roundTripIsl.reversed].each { isl ->
+            assert !database.getIslRoundTripStatus(isl)
+        }
+
+        when: "Install ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.installSwitchRules(dstSw.dpId, InstallRulesAction.INSTALL_ROUND_TRIP_LATENCY)
+        isRoundTripRuleDeleted = false
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.empty
+        }
+
+        then: "The round trip ISL is still FAILED"
```

Comment:
why? Also don't really understand what are you testing here if you will again remove this rule on next step

Reply:
deleted